var/spack/repos/builtin/packages/mapserver/package.py

```diff
@@ -0,0 +1,96 @@
+# Copyright 2013-2018 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+import os
+
+
+class Mapserver(CMakePackage):
+    """MapServer is an Open Source geographic data rendering engine written
+       in C. Beyond browsing GIS data, MapServer allows you create
+       "geographic image maps", that is, maps that can direct users
+       to content"""
+
+    homepage = "http://www.mapserver.org/"
+    url      = "http://download.osgeo.org/mapserver/mapserver-7.2.1.tar.gz"
+
+    version('7.2.1', sha256='9459a7057d5a85be66a41096a5d804f74665381186c37077c94b56e784db6102')
+
+    depends_on('libpng')
+    depends_on('freetype')
+    depends_on('jpeg')
+    depends_on('zlib')
+    depends_on('proj')
+    depends_on('curl', when='+curl')
+    depends_on('geos')
+    depends_on('libxml2')
+    depends_on('gdal')
+    depends_on('swig', type='build')
+    depends_on('python', when='+python')
+    depends_on('ruby', when='+ruby')
+    depends_on('java', when='+java')
+    depends_on('perl', when='+perl')
+
+    variant('python', default=False)
+    variant('curl', default=False)
+    variant('ruby', default=False)
+    variant('java', default=False)
+    variant('perl', default=False)
```

Comment:
Move the variants above the dependencies. Also, each variant needs a description.